packages/main/src/ExpandableText.ts

```diff
@@ -0,0 +1,196 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import i18n from "@ui5/webcomponents-base/dist/decorators/i18n.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import { isPhone } from "@ui5/webcomponents-base/dist/Device.js";
+import Text from "./Text.js";
+import Link, { type LinkAccessibilityAttributes } from "./Link.js";
+import ResponsivePopover from "./ResponsivePopover.js";
+import Button from "./Button.js";
+import ExpandableTextOverflowMode from "./types/ExpandableTextOverflowMode.js";
+import type TextEmptyIndicatorMode from "./types/TextEmptyIndicatorMode.js";
+import {
+	EXPANDABLE_TEXT_SHOW_LESS,
+	EXPANDABLE_TEXT_SHOW_MORE,
+	EXPANDABLE_TEXT_CLOSE,
+	EXPANDABLE_TEXT_SHOW_LESS_POPOVER_ARIA_LABEL,
+	EXPANDABLE_TEXT_SHOW_MORE_POPOVER_ARIA_LABEL,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Template
+import ExpandableTextTemplate from "./generated/templates/ExpandableTextTemplate.lit.js";
+
+// Styles
+import ExpandableTextCss from "./generated/themes/ExpandableText.css.js";
+
+/**
+ * @class
+ *
+ * ### Overview
+ *
+ * The `ui5-expandable-text` component allows displaying a large body of text in a small space. It provides an "expand/collapse" functionality, which shows/hides potentially truncated text.
+ *
+ * ### Usage
+ *
+ * #### When to use:
+ * - To accommodate long texts in limited space, for example in list items or table cell texts
```

Comment:
To accommodate long texts in limited space, for example in list items or table cell texts **or forms**

Reply:
Done